File: 1013-fibonacci-number/fibonacci-number.java
class Solution {
  private int[] memo;
  public int fib(int n) {
    memo = new int[n + 1];
    return fibHelper(n);
  }
  private int fibHelper(int n) {
    if (n < 2) return n;
    if (memo[n] != 0) return memo[n];
    memo[n] = fibHelper(n - 1) + fibHelper(n - 2);
    return memo[n];
  }
}

File: 1013-fibonacci-number/fibonacci-number.rs
impl Solution {
  pub fn fib(n: i32) -> i32 {
    if n < 2 {
      return n;
    }
    let mut memo = vec![0; (n + 1) as usize];
    memo[0] = 0;
    memo[1] = 1;
    for i in 2..=n as usize{
      memo[i] = memo[i - 1] + memo[i - 2];
    }
    memo[n as usize]
  }
}
